code_sender/chrome/chrome-rstudio.applescript

```diff
@@ -1,17 +1,31 @@
 on run argv
     set cmd to item 1 of argv
     tell application "Google Chrome"
-        set URL of front window's active tab to "javascript:{" & "
-            var input = document.getElementById('rstudio_console_input');
-            var textarea = input.getElementsByTagName('textarea')[0];
-            textarea.value += \"" & cmd & "\";
-            var e = document.createEvent('KeyboardEvent');
-            e.initKeyboardEvent('input');
-            textarea.dispatchEvent(e);
-            var e = document.createEvent('KeyboardEvent');
-            e.initKeyboardEvent('keydown');
-            Object.defineProperty(e, 'keyCode', {'value' : 13});
-            input.dispatchEvent(e);
-        " & "}"
+        repeat with w in (windows)
+            set j to 0
+            repeat with t in (tabs of w)
+                set j to j + 1
+                if title of t contains "RStudio" then
+                    set (active tab index of w) to j
+                    set index of w to 1
+
+                    set URL of front window's active tab to "javascript:{" & "
+                        var input = document.getElementById('rstudio_console_input');
+                        var textarea = input.getElementsByTagName('textarea')[0];
+                        textarea.value += \"" & cmd & "\";
+                        var e = document.createEvent('KeyboardEvent');
+                        e.initKeyboardEvent('input');
+                        textarea.dispatchEvent(e);
+                        var e = document.createEvent('KeyboardEvent');
+                        e.initKeyboardEvent('keydown');
+                        Object.defineProperty(e, 'keyCode', {'value' : 13});
+                        input.dispatchEvent(e);
+                    " & "}"
+
+                    tell application "Sublime Text" to activate
```

Comment:
Why do we need to reactivate sublime? I thought that we never change focus

Reply:
In line number 9 we activate the Chrome window where RStudio is open. This is why we need to reactivate sublime. The activation of that Chrome window is necessary because we cannot send code if it's minimized.

Reply:
could we somehow check if the active window in RStudio so we don't have to focus and refocus every time we send something?